lib/_stream_readable.js

```diff
@@ -72,7 +72,7 @@ function ReadableState(options, stream) {
   this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
 
   // cast to ints.
-  this.highWaterMark = ~~this.highWaterMark;
+  this.highWaterMark = Math.floor(this.highWaterMark);
```

Comment:
This is correct, but not an actual problem. According to @mcollina:

> Any negative value is interpreted by the stream machinery as a 0

Therefore, it does not matter whether the value is `Math.floor(-233.5) = -234` or `~~(-233.5) = -233`, not even in the special case of `Math.floor(-0.5) = -1` whilst `~~(-0.5) = 0`.

We are [planning to throw on negative values](https://github.com/nodejs/node/pull/12593#issuecomment-299124070) anyway.